tidy3d/plugins/sweep/result.py

```diff
@@ -0,0 +1,322 @@
+
+        if self.fn_source != other.fn_source:
+            raise ValueError("Can't combine results, function sources don't match.")
+
+        if self.output_names != other.output_names:
+            raise ValueError("Can't combine results, output names don't match.")
+
+        if self.dims != other.dims:
+            raise ValueError("Can't combine results, dimensions don't match.")
+
+        def combine_tuples(tuple1: tuple, tuple2: tuple):
+            """Combine two tuples together if not None."""
+            if tuple1 is None and tuple2 is None:
+                return None
+            if (tuple1 is None) != (tuple2 is None):
+                raise ValueError("Can't combine data where one only one field is `None`.")
+            return list(tuple1) + list(tuple2)
+
+        task_ids = combine_tuples(self.task_ids, other.task_ids)
+        coords = combine_tuples(self.coords, other.coords)
+        values = combine_tuples(self.values, other.values)
+
+        return Result(
+            dims=self.dims,
+            coords=coords,
+            values=values,
+            output_names=self.output_names,
+            fn_source=self.fn_source,
+            task_ids=task_ids,
+        )
+
+    def __add__(self, other):
+        """Special syntax for design_result1 + design_result2."""
+        return self.combine(other)
+
+    def get_index(self, fn_args: Dict[str, float]) -> int:
+        """Get index into the data for a specific set of arguments."""
+
+        key_list = list(self.coords)
+        arg_key = tuple(fn_args[dim] for dim in self.dims)
+        return key_list.index(arg_key)
+
+    def delete(self, fn_args: Dict[str, float]) -> Result:
+        """Delete a specific set of arguments from the result."""
+
+        index = self.get_index(fn_args)
+
+        new_coords = list(self.coords)
+        new_values = list(self.values)
+
+        new_coords.pop(index)
+        new_values.pop(index)
+
+        return self.updated_copy(values=new_values, coords=new_coords)
+
+    def add(self, fn_args: Dict[str, float], value: Any) -> Result:
+        """Add a specific argument and value the result."""
+
+        new_values = list(self.values) + [value]
+        new_coords = list(self.coords) + [tuple(fn_args[dim] for dim in self.dims)]
+
+        return self.updated_copy(values=new_values, coords=new_coords)
+
+    def hexbin(self, x: str, y: str, C: str, **hexbin_kwargs):
+        """Visualize the stored data with `x` and `y` axes referenced by dimension keys."""
+
+        # https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.plot.hexbin.html
+
+        df = self.data_frame
+
+        if "gridsize" not in hexbin_kwargs:
+            hexbin_kwargs["gridsize"] = len(np.unique(df[x]))
+
+        im = df.plot.hexbin(x=x, y=y, C=C, **hexbin_kwargs)
+        return im
+
+    def scatter(self, x: str, y: str, s: str = None, c: str = None, **scatter_kwargs):
```

Comment:
Maybe rename it to `plot` to be consistent with the rest of our API?

Reply:
Same issue as the hexbin? I think I will also remove `scatter` and more generally just recommend people use `DataFrame` for all plotting needs.